addons/survey/static/src/js/survey_form.js

```diff
@@ -408,9 +409,76 @@ publicWidget.registry.SurveyFormWidget = publicWidget.Widget.extend(SurveyPreloa
+            submitPromise = Promise.resolve(result)
+            if (Object.keys(correctAnswers).length) {
+                this._showCorrectAnswers(correctAnswers, submitPromise, options);
+                return;
+            }
+        }
         this._nextScreen(submitPromise, options);
     },
 
+    _showCorrectAnswers: function(correctAnswers, submitPromise, options) {
+        const inactiveQuestionIds = this.options.sessionInProgress ? [] : this._getInactiveConditionalQuestionIds();
+        const activeQuestionIds = Object.keys(correctAnswers).filter(
+            questionId => !inactiveQuestionIds.includes(parseInt(questionId))
+        );
+        // Display correct answers
+        activeQuestionIds.forEach(questionId => {
+            const correctAnswer = correctAnswers[questionId];
+            const questionWrapper = document.querySelector(`.js_question-wrapper[id="${questionId}"]`);
+            const answerWrapper = questionWrapper.querySelector('.o_survey_answer_wrapper');
+            const questionType = questionWrapper.querySelector('[data-question-type]').dataset.questionType;
+
+            if (['numerical_box', 'date', 'datetime'].includes(questionType)) {
+                const input = answerWrapper.querySelector('input');
+                let value;
+                if (questionType == 'numerical_box') {
+                    value = input.valueAsNumber;
+                } else if (questionType == 'datetime') {
+                    const datetime = parseDateTime(input.value);
+                    value = datetime ? datetime.setZone("utc").toFormat("MM/dd/yyyy HH:mm:ss", { numberingSystem: "latn" }) : '';
+                } else {
+                    value = input.value;
+                }
+                const isCorrect = value == correctAnswer;
+                answerWrapper.classList.add(`bg-${isCorrect ? 'success' : 'danger'}`);
+            }
+            else if (['simple_choice_radio', 'multiple_choice'].includes(questionType)) {
+                answerWrapper.querySelectorAll('.o_survey_choice_btn').forEach((button) => {
+                    const answerId = button.querySelector('input').value;
+                    const isCorrect = correctAnswer.includes(parseInt(answerId));
+                    button.classList.add(`bg-${isCorrect ? 'success' : 'danger'}`, 'text-white');
+                    if (!isCorrect) {
+                        let fromIcon = 'fa-check-circle';
+                        let toIcon = 'fa-times-circle';
+                        if (questionType == 'multiple_choice') {
+                            fromIcon = 'fa-check-square';
+                            toIcon = 'fa-times-rectangle'; // fa-times-square doesn't exist in fontawesome 4.7
+                        }
+                        button.querySelector(`i.${fromIcon}`)?.classList.replace(fromIcon, toIcon);
+                    }
+                });
+            }
+        });
```

Comment:
As it is getting long, I would separate survey-level from question-level manipulations and make another method for this.
It would make it more obvious that this method 1. Shows answers and 2. prepares "nextScreen"
(An alternative would be to put all survey-level code in _submitForm but it is also very long...)